### PR TITLE
Library feature

### PR DESCRIPTION
**Pull request by ikraamg**
Update after second review:
- Form resets on submit.
- Message displays not alert.

Update after first review:
- Separated Logic and Dom manipulation JS files.
- Removed commented code.
- Added form controls.
- Refactored.

Good Day,

Please find the [Readme](https://github.com/alishabab/javascript-library/blob/library-feature/README.md) and the [Live Link](https://rawcdn.githack.com/alishabab/javascript-library/38488039fd980e7d107fbeae2b2ef9b0447403f2/index.html).

In addition to the requirements, we followed clean code principles by:
- Using classes from ES5+.
- Using descriptive function and variable names.
- Separating tasks into separate functions.

Thanks!

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+     <h1 class="text-center main-logo">My Library</h1>
+    <!-- <label for="title">Title</label>
+    <input type="text" name="title" id="title" />
+    <label for="author">author</label>
+    <input type="text" name="author" id="author" />
+    <label for="pages">pages</label>
+    <input type="text" name="pages" id="pages" />
+    <label for="read">read</label>
+    <input type="text" name="read" id="read" />
+    <button type="submit" id= 'create'>Create Book</button> -->
```
Comment (bellom) [inline index.html]: . It’s better to delete commented code. If you need to review a previous version of your code, use git history. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

File: library.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  <input type="number" name="pages" id="pages" />
+  <br>
+  <label for="read">Read</label>
+
+  <input type="checkbox" name="read" id="read" />
+  <br>
+  <button type="submit" id= 'create'>Create Book</button>`;
+
+  document.querySelector('#create').addEventListener('click', addBookToLibrary);
+};
```
Comment (bellom) [inline library.js]: Let's separate the DOM from application logic, you can create another file for DOM.

image

https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59623451/

File: library.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  <br>
+  <label for="pages">Pages</label>
+
+  <input type="number" name="pages" id="pages" />
+  <br>
+  <label for="read">Read</label>
+
+  <input type="checkbox" name="read" id="read" />
+  <br>
+  <button type="submit" id= 'create'>Create Book</button>`;
```
Comment (bellom) [inline library.js]: You need to do form controls to make sure empty forms cannot be submitted
image

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi Team :clap: 

### Awesome job so far!! :+1: 

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 
 
**Kindly check out my comments below;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

File: index.html [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      <br />
+      <label for="pages">Pages</label>
+
+      <input type="number" name="pages" id="pages" />
+      <br />
+      <label for="read">Read</label>
+
+      <input type="checkbox" name="read" id="read" />
+      <br />
+      <button type="submit" id="create">Create Book</button>
```
Comment (bellom) [inline index.html]: Opps, i don't think you were having this issue before..... 🤔 
Kindly clear the form once the user creates a book.
(I need to clear the form manually to add a new book.)
image

File: js/dom.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+  if (title.length > 0 && author.length > 0 && pages.length > 0) {
+    const newbook = new Book(title, author, pages, read);
+
+    myLibrary.push(newbook);
+
+    renderForm();
+    render();
+  } else {
+    // eslint-disable-next-line no-alert
+    window.alert('Title/Author/Pages must not be empty');
```
Comment (bellom) [inline js/dom.js]: Opps 🤦‍♂️ , This should be production code, we shouldn't have `alert`, please remove the `alert`.
If you don't want to pop-up new div message, I think you can make use of `required` on the inputs tags.

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi Team  :clap: 

**Great work making `almost` all the necessary corrections from the previous review.**

**Kindly check out my comments below;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Review by fatymahmed — APPROVED** [review]
## Hi @ikraamg @alishabab,
Good job implementing the changes 🎉🎉

## Status: Approved ✔️
_You may go ahead and merge this PR_

**Pull request merged**